merkle/log_verifier_test.go

```diff
@@ -17,402 +17,330 @@ package merkle
+		dh("d37ee418976dd95753c1c73862b9398fa2a2cf9b4ff0fdfe8b30cd95209614b7", 32),
+		dh("4e3bbb1f7b478dcfe71fb631631519a3bca12c9aefca1612bfce4c13a86264d4", 32),
+		dh("76e67dadbcdf1e10e1b74ddc608abd2f98dfb16fbce75277b5232a127f2087ef", 32),
+		dh("ddb89be403809e325750d3d263cd78929c2942b7942a34b77e122c9594a74c8c", 32),
+		dh("5dc9da79a70659a9ad559cb701ded9a2ab9d823aad2f4960cfe370eff4604328", 32),
+	}
+
+	leaves = [][]byte{
+		dh("", 0),
+		dh("00", 1),
+		dh("10", 1),
+		dh("2021", 2),
+		dh("3031", 2),
+		dh("40414243", 4),
+		dh("5051525354555657", 8),
+		dh("606162636465666768696a6b6c6d6e6f", 16),
 	}
 )
 
-func verifierCheck(v *LogVerifier, leafIndex, treeSize int64, proof [][]byte, root, leafHash []byte) error {
-	// Verify original inclusion proof
-	got, err := v.RootFromInclusionProof(leafIndex, treeSize, proof, leafHash)
-	if err != nil {
-		return err
-	}
-	if want := root; !bytes.Equal(got, want) {
-		return fmt.Errorf("got root:\n%s\nexpected:\n%s", shortHash(got), shortHash(want))
-	}
-	if err := v.VerifyInclusionProof(leafIndex, treeSize, proof, root, leafHash); err != nil {
-		return err
-	}
+// inclusionProbe is a parameter set for inclusion proof verification.
+type inclusionProbe struct {
+	leafIndex int64
+	treeSize  int64
+	root      []byte
+	leafHash  []byte
+	proof     [][]byte
 
-	// Wrong leaf index
-	if err := v.VerifyInclusionProof(leafIndex-1, treeSize, proof, root, leafHash); err == nil {
-		return errors.New("incorrectly verified against leafIndex - 1")
-	}
-	if err := v.VerifyInclusionProof(leafIndex+1, treeSize, proof, root, leafHash); err == nil {
-		return errors.New("incorrectly verified against leafIndex + 1")
-	}
-	if err := v.VerifyInclusionProof(leafIndex^2, treeSize, proof, root, leafHash); err == nil {
-		return errors.New("incorrectly verified against leafIndex ^ 2")
-	}
+	desc string
+}
 
-	// Wrong tree height
-	if err := v.VerifyInclusionProof(leafIndex, treeSize*2, proof, root, leafHash); err == nil {
-		return errors.New("incorrectly verified against treeSize * 2")
-	}
-	if err := v.VerifyInclusionProof(leafIndex, treeSize/2, proof, root, leafHash); err == nil {
-		return errors.New("incorrectly verified against treeSize / 2")
-	}
+// consistencyProbe is a parameter set for consistency proof verification.
+type consistencyProbe struct {
+	snapshot1 int64
+	snapshot2 int64
+	root1     []byte
+	root2     []byte
+	proof     [][]byte
 
-	// Wrong leaf
-	if err := v.VerifyInclusionProof(leafIndex, treeSize, proof, root, []byte("WrongLeaf")); err == nil {
-		return errors.New("incorrectly verified against WrongLeaf")
-	}
+	desc string
+}
 
-	// Wrong root
-	if err := v.VerifyInclusionProof(leafIndex, treeSize, proof, sha256EmptyTreeHash, leafHash); err == nil {
-		return errors.New("incorrectly verified against empty root hash")
-	}
+func corruptInclVerification(leafIndex, treeSize int64, proof [][]byte, root, leafHash []byte) []inclusionProbe {
```

Comment:
rename to `corruptInclusionProof`, the `...Verification` is a bit misleading?
(same for `corruptConsistencyProof` below, if you think it's a good idea)

Reply:
Done.